WooCommerce/Classes/ViewRelated/Orders/Order Creation/CustomerSection/CustomerSelectorViewController.swift

```diff
@@ -8,11 +8,27 @@ import Yosemite
 final class CustomerSelectorViewController: UIViewController, GhostableViewController {
     private var searchViewController: UIViewController?
     private let siteID: Int64
+    private let onCustomerSelected: (Customer) -> Void
+    private let viewModel: CustomerSelectorViewModel
+
+    /// Notice presentation handler
+    ///
+    private var noticePresenter: NoticePresenter = DefaultNoticePresenter()
+
+    private lazy var activityIndicator: UIActivityIndicatorView = {
+        let indicator = UIActivityIndicatorView(style: .medium)
+        indicator.hidesWhenStopped = true
+        indicator.translatesAutoresizingMaskIntoConstraints = false
+        return indicator
+    }()
 
     lazy var ghostTableViewController = GhostTableViewController(options: GhostTableViewOptions(cellClass: TitleAndSubtitleAndStatusTableViewCell.self))
 
-    init(siteID: Int64) {
+    init(siteID: Int64,
+         onCustomerSelected: @escaping (Customer) -> Void) {
+        viewModel = CustomerSelectorViewModel(siteID: siteID, onCustomerSelected: onCustomerSelected)
```

Comment:
Yeah, good point. In this case, we let the view controller create its own view model because we are not going to test it., that way we keep the initializer cleaner.  You can see that we use this approach in several view controllers along the app.

Reply:
That makes sense, thanks!